File: typescript/src/@stack/stack.ts
export class Stack<T> {
  private items: Array<T> = new Array<T>();

  public clear(): void {
    this.items = new Array<T>();
  }

  public push(item: T): void {
    this.items.push(item);
  }

  public pop(): void {
    this.items.pop();
  }

  public peek(): T {
    return this.items[this.items.length - 1];
  }

  public isEmpty(): boolean {
    return this.items.length === 0;
  }

  public size(): number {
    return this.items.length;
  }

  public print(): void {
    console.log("Stack: ", this.items);
  }
}

File: golang/stack/stack.go
package stack

type Stack[T any] struct {
	elements []T
}

func New[T any]() *Stack[T] {
	return &Stack[T]{}
}

func (s *Stack[T]) Clear() {
	s.elements = []T{}
}

func (s *Stack[T]) Push(e T) {
	s.elements = append(s.elements, e)
}

func (s *Stack[T]) Pop() T {
	if len(s.elements) == 0 {
		var zero T
		return zero
	}
	lastIndex := len(s.elements) - 1
	element := s.elements[lastIndex]
	s.elements = s.elements[:lastIndex]
	return element
}

func (s *Stack[T]) Peek() T {
	if len(s.elements) == 0 {
		var zero T
		return zero
	}
	return s.elements[len(s.elements)-1]
}

func (s *Stack[T]) IsEmpty() bool {
	return len(s.elements) == 0
}

func (s *Stack[T]) Size() int {
	return len(s.elements)
}

func (s *Stack[T]) Print() {
	for i := len(s.elements) - 1; i >= 0; i-- {
		println(s.elements[i])
	}
}
